### PR TITLE
Configure testing, Travis CI, SCSS, and Heroku build.

### PR DESCRIPTION
**Pull request by tmikeschu**
One reviewer needed: @DavidKnott @ethanbennett @ski-climb @case-eee 

No real functionality added here, just the foundation for Enzyme/Jest, Travis CI / Heroku, and SASS.

Add config for Enzyme and Jest
Update favicon and index.html boilerplate
Move App parts to App/
Set up SCSS
Change path dependencies
Add Travis CI config
Comment out blockchain related logic
  (Mutes linting problems related to Travis CI)

[#142657615]

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,6 +3,11 @@
   "version": "0.1.0",
   "private": true,
   "devDependencies": {
+    "enzyme": "^2.8.0",
```
Comment (DavidKnott) [inline package.json]: What does enzyme do?

**Comment by DavidKnott** [comment]
@tmikeschu Looks good to me!

File: package.json (hunk 1 above)
Comment (tmikeschu) [inline package.json]: @DavidKnott Enzyme is our way to do Capybara/browser type stuff. You can `shallow`, `mount`, and `render` a component and run assertions on that state. The [docs](http://airbnb.io/enzyme/docs/api/) are super helpful.

**Comment by DavidKnott** [comment]
Merging now

File: package.json (hunk 1 above)
Comment (DavidKnott) [inline package.json]: Great find! I'm excited to dive into react TDD.

**Pull request merged**